demos/healthcare_assistant_demo/main.py

```diff
@@ -0,0 +1,312 @@
+    # load llama model and its tokenizer
+    if not model_path.exists():
+        chat_model = OVModelForCausalLM.from_pretrained(model_name, export=True, compile=False, load_in_8bit=False,
+                                                        token=token)
+
+        quant_config = OVWeightQuantizationConfig(bits=4, sym=False, ratio=0.8)
+        config = OVConfig(quantization_config=quant_config)
+
+        quantizer = OVQuantizer.from_pretrained(chat_model, task="text-generation")
+        quantizer.quantize(save_directory=model_path, weights_only=True, ov_config=config)
+
+        chat_tokenizer = AutoTokenizer.from_pretrained(model_name, token=token)
+        chat_tokenizer.save_pretrained(model_path)
+
+    return OpenVINOLLM(context_window=2048, model_name=str(model_path), tokenizer_name=str(model_path),
+                       max_new_tokens=512, device_map=device, model_kwargs={"ov_config": ov_config},
+                       generate_kwargs={"do_sample": True, "temperature": 0.7, "top_k": 50, "top_p": 0.95},
+                       messages_to_prompt=get_conversation)
+
+
+def load_embedding_model(model_name: str) -> OpenVINOEmbedding:
+    model_path = MODEL_DIR / model_name
+
+    if not model_path.exists():
+        embedding_model = OVModelForFeatureExtraction.from_pretrained(model_name, export=True, device="AUTO")
+        embedding_model.save_pretrained(model_path)
+        embedding_tokenizer = AutoTokenizer.from_pretrained(model_name)
+        embedding_tokenizer.save_pretrained(model_path)
+
+    return OpenVINOEmbedding(str(model_path))
+
+
+def load_chat_models(chat_model_name: str, embedding_model_name: str, auth_token: str = None) -> None:
+    global ov_llm, ov_chat_engine, ov_embedding
+    ov_embedding = load_embedding_model(embedding_model_name)
+    ov_llm = load_chat_model(chat_model_name, auth_token)
+
+    ov_chat_engine = SimpleChatEngine.from_defaults(llm=ov_llm, system_prompt=SYSTEM_CONFIGURATION)
+
+
+def load_file(file_path: Path) -> Document:
+    ext = file_path.suffix
+    if ext == ".pdf":
+        reader = PDFReader()
+        return reader.load_data(file_path)[0]
+    elif ext == ".txt":
+        with open(file_path) as f:
+            content = f.read()
+            return Document(text=content, metadata={"file_name": file_path.name})
+    else:
+        raise ValueError(f"{ext} file is not supported for now")
+
+
+def load_context(file_path: str) -> str:
+    global ov_chat_engine
+    if not file_path:
+        ov_chat_engine = SimpleChatEngine.from_defaults(llm=ov_llm, system_prompt=SYSTEM_CONFIGURATION)
+        return "No report loaded"
+
+    document = load_file(Path(file_path))
+
+    Settings.embed_model = ov_embedding
+    index = VectorStoreIndex.from_documents([document])
+    ov_chat_engine = index.as_chat_engine(llm=ov_llm, chat_mode=ChatMode.CONTEXT, system_prompt=SYSTEM_CONFIGURATION)
+
+    return "Report loaded!"
+
+
+def get_conversation(messages: Sequence[ChatMessage]) -> str:
```

Comment:
I added this to get rid of these strange "assistant" keywords. How to implement completion_to_prompt? What should it return? Any examples?

Reply:
def completion_to_prompt(completion):
    return f"<|begin_of_text|><|start_header_id|>system<|end_header_id|><|eot_id|><|start_header_id|>user<|end_header_id|>{completion}<|eot_id|><|start_header_id|>assistant<|end_header_id|>"


llm = OpenVINOLLM(
    model_id_or_path=str(llm_model_path),
    context_window=3900,
    max_new_tokens=1000,
    model_kwargs={"ov_config": ov_config},
    device_map=llm_device.value,
    completion_to_prompt=completion_to_prompt,
)

Reply:
Hmm, I cannot see any benefit of providing this template manually instead of using a template from the tokenizer. Could you explain why that approach would be better?

Reply:
Here we are talking about `completion_to_prompt` instead of `messages_to_prompt` which can not be applied with a template from tokenizer

Reply:
`completion_to_prompt `is not used unless you run `complete(`) function